File: TP2 - Pizzas.API v2/Services/PizzaService.cs
using System.Collections.Generic;
using System.Linq;
using System.Data.SqlClient;
using Dapper;
using Pizzas.API.Models;
using Pizzas.API.Utils;

namespace Pizzas.API.Services
{
    public static class PizzaService
    {
        public static List<Pizza> GetAll()
        {
            string sqlQuery = $"SELECT Id, Nombre, LibreGluten, Importe, Descripcion FROM Pizzas";
            List<Pizza> returnList;

            returnList = new List<Pizza>();
            using (SqlConnection db = BD.GetConnection())
            {
                returnList = db.Query<Pizza>(sqlQuery).ToList();
            }

            return returnList;
        }
        public static Pizza GetById(int id)
        {
            string sqlQuery = $"SELECT Id, Nombre, LibreGluten, Importe, Descripcion FROM Pizzas WHERE Id = {id}";
            Pizza returnEntity = null;
            using (SqlConnection db = BD.GetConnection())
            {
                returnEntity = db.QueryFirstOrDefault<Pizza>(sqlQuery);
            }
            return returnEntity;
        }
        public static int Add(Pizza pizza)
        {
            string sqlQuery = $"INSERT INTO Pizzas (Nombre, LibreGluten, Importe, Descripcion) VALUES (@nombre, @libreGluten, @importe, @descripcion) ";
            int intRowsAffected = 0;
            using (SqlConnection db = BD.GetConnection())
            {
                intRowsAffected = db.Execute(sqlQuery, new {
                    nombre= pizza.Nombre,
                    libreGluten= pizza.LibreGluten,
                    importe= pizza.Importe,
                    descripcion= pizza.Descripcion
                });
            }
            return intRowsAffected;
        }

        public static int UpdateById(Pizza pizza)
        {
            int intRowsAffected = 0;
            string sqlQuery = $"UPDATE Pizzas SET Nombre = @nombre, LibreGluten = @libreGluten, Importe = @importe, Descripcion = @descripcion WHERE Id = @pizzaId";
            using (SqlConnection db = BD.GetConnection())
            {
                intRowsAffected = db.Execute(sqlQuery, new {
                    pizzaId = pizza.Id,
                    nombre= pizza.Nombre,
                    libreGluten= pizza.LibreGluten,
                    importe= pizza.Importe,
                    descripcion= pizza.Descripcion
                });
            }
            return intRowsAffected;
        }
        public static int DeleteById(int id)
        {
            string sqlQuery = $"DELETE FROM Pizzas WHERE Id = {id}";
            int intRowsAffected = 0;
            using (SqlConnection db = BD.GetConnection())
            {
                intRowsAffected = db.Execute(sqlQuery);
            }
            return intRowsAffected;
        }
    }
}

File: Node & MSSql/src/services/pizzaService.js
import config from "../../dbconfig.js";
import sql from "mssql";

export default class PizzaService {
    getAll = async () => {
        try {
            const pool = await sql.connect(config);
            const result = await pool.request().query("SELECT * FROM Pizzas");
            return result.recordset;
        } catch (err) {
            console.log(err);
        }
    }
    getById = async (id) => {
        try {
            const pool = await sql.connect(config);
            const result = await pool.request()
                .input("id", sql.Int, id)
                .query("SELECT * FROM Pizzas WHERE id = @id");
            return result.recordset;
        } catch (err) {
            console.log(err);
        }
    }
    insert = async (pizza) => {
        try {
            const pool = await sql.connect(config);
            const result = await pool.request()
                .input("nombre", sql.VarChar, pizza.nombre)
                .input("libreGluten", sql.Bit, pizza.libreGluten)
                .input("importe", sql.Decimal(10, 2), pizza.importe)
                .input("descripcion", sql.VarChar, pizza.descripcion)
                .query("INSERT INTO Pizzas (nombre, libreGluten, importe, descripcion) VALUES (@nombre, @libreGluten, @importe, @descripcion)");
            return result.rowsAffected;
        } catch (err) {
            console.log(err);
        }
    }
    update = async (pizza) => {
        try {
            const pool = await sql.connect(config);
            const result = await pool.request()
                .input("id", sql.Int, pizza.id)
                .input("nombre", sql.VarChar, pizza.nombre)
                .input("libreGluten", sql.Bit, pizza.libreGluten)
                .input("importe", sql.Decimal(10, 2), pizza.importe)
                .input("descripcion", sql.VarChar, pizza.descripcion)
                .query("UPDATE Pizzas SET nombre = @nombre, libreGluten = @libreGluten, importe = @importe, descripcion = @descripcion WHERE id = @id");
            return result.rowsAffected;
        } catch (err) {
            console.log(err);
        }
    }
    deleteById = async (id) => {
        try {
            const pool = await sql.connect(config);
            const result = await pool.request()
                .input("id", sql.Int, id)
                .query("DELETE FROM Pizzas WHERE id = @id");
            return result.rowsAffected;
        } catch (err) {
            console.log(err);
        }

    }
}